### ISSUE TITLE
Unable to add profile picture: 400 error while cropping

### ISSUE THREAD

**Issue by penguin86** [nextcloud/server#28556]
### Steps to reproduce
1. Go to user's settings
2. Set a profile picture
3. Set the cropping margins when asked and submit 

### Expected behaviour
The cropped picture should be set as user profile picture

### Actual behaviour
The loading spinner is shown indefinitely and no profile picture is set

### Server configuration

**Operating system:**
Debian 11

**Web server:**
Server version: Apache/2.4.48 (Debian)
Server built:   2021-08-12T11:51:47

**Database:**
MariaDB 10.5.11 database server

**PHP version:**
PHP 7.4.21

**Nextcloud version:** (see Nextcloud admin page)
Nextcloud 21.0.4

**Updated from an older Nextcloud/ownCloud or fresh install:**
Updated from 21.x

**Where did you install Nextcloud from:**
Manual installation as documented in https://docs.nextcloud.com/server/latest/admin_manual/installation/source_installation.html

**Signing status:**
<details>
<summary>Signing status</summary>

```
No errors have been found.
```
</details>

**List of activated apps:**
<details>
<summary>App list</summary>

```
Enabled:
  - accessibility: 1.7.0
  - activity: 2.14.3
  - bookmarks: 4.4.0
  - calendar: 2.3.2
  - cloud_federation_api: 1.4.0
  - contacts: 4.0.1
  - dashboard: 7.1.0
  - dav: 1.17.1
  - deck: 1.4.3
  - extract: 1.3.2
  - federatedfilesharing: 1.11.0
  - files: 1.16.0
  - files_markdown: 2.3.4
  - files_pdfviewer: 2.1.0
  - files_rightclick: 1.0.0
  - files_sharing: 1.13.1
  - files_trashbin: 1.11.0
  - files_versions: 1.14.0
  - files_videoplayer: 1.10.0
  - groupfolders: 9.0.2
  - guests: 2.0.1
  - integration_google: 1.0.2
  - logreader: 2.6.0
  - lookup_server_connector: 1.9.0
  - maps: 0.1.9
  - notes: 4.1.1
  - notifications: 2.9.0
  - oauth2: 1.9.0
  - password_policy: 1.11.0
  - passwords: 2021.7.23
  - photos: 1.3.0
  - privacy: 1.5.0
  - provisioning_api: 1.11.0
  - recommendations: 1.0.0
  - serverinfo: 1.11.0
  - settings: 1.3.0
  - sharebymail: 1.11.0
  - spreed: 11.3.1
  - survey_client: 1.9.0
  - tasks: 0.14.1
  - text: 3.2.0
  - twofactor_backupcodes: 1.10.0
  - twofactor_nextcloud_notification: 3.2.1
  - updatenotification: 1.11.0
  - viewer: 1.5.0
  - weather_status: 1.1.0
  - workflowengine: 2.3.0
Disabled:
  - admin_audit
  - comments
  - contactsinteraction
  - encryption
  - federation
  - files_external
  - firstrunwizard
  - nextcloud_announcements
  - qownnotesapi
  - support
  - systemtags
  - theming
  - user_ldap
  - user_status
```
</details>

**Nextcloud configuration:**
<details>
<summary>Config report</summary>

```{
    "system": {
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
            "localhost",
            "192.168.1.99",
            "OMITTED_FOR_PRIVACY.com"
        ],
        "datadirectory": "***REMOVED SENSITIVE VALUE***",
        "dbtype": "mysql",
        "version": "21.0.4.1",
        "overwrite.cli.url": "https:\/\/OMITTED_FOR_PRIVACY.com\/nextcloud",
        "dbname": "***REMOVED SENSITIVE VALUE***",
        "dbhost": "***REMOVED SENSITIVE VALUE***",
        "dbport": "",
        "dbtableprefix": "oc_",
        "mysql.utf8mb4": true,
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "installed": true,
        "instanceid": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpmode": "smtp",
        "mail_smtphost": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpport": "465",
        "mail_smtpsecure": "ssl",
        "mail_smtptimeout": 30,
        "mail_smtpauth": 1,
        "mail_smtpauthtype": "LOGIN",
        "mail_smtpname": "***REMOVED SENSITIVE VALUE***",
        "mail_smtppassword": "***REMOVED SENSITIVE VALUE***",
        "maintenance": false,
        "mail_from_address": "***REMOVED SENSITIVE VALUE***",
        "mail_sendmailmode": "smtp",
        "mail_domain": "***REMOVED SENSITIVE VALUE***",
        "memcache.local": "\\OC\\Memcache\\APCu",
        "memcache.distributed": "\\OC\\Memcache\\Redis",
        "redis": {
            "host": "***REMOVED SENSITIVE VALUE***",
            "port": 6379
        },
        "theme": "",
        "loglevel": 2
    }
}
```
</details>

**Are you using external storage, if yes which one:** local/smb/sftp/...
No

**Are you using encryption:** yes/no
No

**Are you using an external user-backend, if yes which one:** LDAP/ActiveDirectory/Webdav/...
No

### Client configuration
**Browser:**
Firefox 78.12.0esr (64 bit)

**Operating system:**
Debian 11 with KDE

### Logs
```
Error | core | OCP\Files\NotPermittedException: /var/www/html/nextcloud/lib/private/Files/SimpleFS/SimpleFile.php - line 150: OC\Files\Node\File->delete()/var/www/html/nextcloud/lib/private/Avatar/UserAvatar.php - line 209: OC\Files\SimpleFS\SimpleFile->delete()/var/www/html/nextcloud/lib/private/Avatar/UserAvatar.php - line 106: OC\Avatar\UserAvatar->remove()/var/www/html/nextcloud/core/Controller/AvatarController.php - line 314: OC\Avatar\UserAvatar->set()/var/www/html/nextcloud/lib/private/AppFramework/Http/Dispatcher.php - line 218: OC\Core\Controller\AvatarController->postCroppedAvatar()/var/www/html/nextcloud/lib/private/AppFramework/Http/Dispatcher.php - line 127: OC\AppFramework\Http\Dispatcher->executeController()/var/www/html/nextcloud/lib/private/AppFramework/App.php - line 157: OC\AppFramework\Http\Dispatcher->dispatch()/var/www/html/nextcloud/lib/private/Route/Router.php - line 302: OC\AppFramework\App::main()/var/www/html/nextcloud/lib/base.php - line 993: OC\Route\Router->match()/var/www/html/nextcloud/index.php - line 37: OC::handleRequest()
-- | -- | --
```

#### Web server error log
<details>
<summary>Web server error log</summary>

```
[Sun Aug 22 10:32:03.681823 2021] [access_compat:error] [pid 275645] [client 192.168.1.56:53974] AH01797: client denied by server configuration: /var/www/html/nextcloud/config
```
Note: I'm not sure the log is relevant to the profile picture upload, as I tried multiple uploads, but I can see a single log line, and [this](https://help.nextcloud.com/t/client-denied-by-server-configuration/76163) seems to suggest it is an error related to Nextcloud desktop client.
</details>

#### Nextcloud log (data/nextcloud.log)
<details>
<summary>Nextcloud log</summary>

```
{"reqId":"RzeLGrxj80IS7XrdoGZx","level":3,"time":"2021-08-22T09:34:07+00:00","remoteAddr":"192.168.1.56","user":"danieleverducci","app":"core","method":"POST","url":"/index.php/avatar/cropped","message":{"Exception":"OCP\\Files\\NotPermittedException","Message":"","Code":0,"Trace":[{"file":"/var/www/html/nextcloud/lib/private/Files/SimpleFS/SimpleFile.php","line":150,"function":"delete","class":"OC\\Files\\Node\\File","type":"->"},{"file":"/var/www/html/nextcloud/lib/private/Avatar/UserAvatar.php","line":209,"function":"delete","class":"OC\\Files\\SimpleFS\\SimpleFile","type":"->"},{"file":"/var/www/html/nextcloud/lib/private/Avatar/UserAvatar.php","line":106,"function":"remove","class":"OC\\Avatar\\UserAvatar","type":"->"},{"file":"/var/www/html/nextcloud/core/Controller/AvatarController.php","line":314,"function":"set","class":"OC\\Avatar\\UserAvatar","type":"->"},{"file":"/var/www/html/nextcloud/lib/private/AppFramework/Http/Dispatcher.php","line":218,"function":"postCroppedAvatar","class":"OC\\Core\\Controller\\AvatarController","type":"->"},{"file":"/var/www/html/nextcloud/lib/private/AppFramework/Http/Dispatcher.php","line":127,"function":"executeController","class":"OC\\AppFramework\\Http\\Dispatcher","type":"->"},{"file":"/var/www/html/nextcloud/lib/private/AppFramework/App.php","line":157,"function":"dispatch","class":"OC\\AppFramework\\Http\\Dispatcher","type":"->"},{"file":"/var/www/html/nextcloud/lib/private/Route/Router.php","line":302,"function":"main","class":"OC\\AppFramework\\App","type":"::"},{"file":"/var/www/html/nextcloud/lib/base.php","line":993,"function":"match","class":"OC\\Route\\Router","type":"->"},{"file":"/var/www/html/nextcloud/index.php","line":37,"function":"handleRequest","class":"OC","type":"::"}],"File":"/var/www/html/nextcloud/lib/private/Files/Node/File.php","Line":138,"CustomMessage":"--"},"userAgent":"Mozilla/5.0 (X11; Linux x86_64; rv:78.0) Gecko/20100101 Firefox/78.0","version":"21.0.4.1"}
```
</details>

#### Browser log
<details>
<summary>Browser log</summary>

```
a) The javascript console log
Nothing useful, only unrelated warnings.

b) The network log

POST | https://OMITTED_FOR_PRIVACY/index.php/avatar/cropped

Request:
crop[x] | "120"
crop[y] | "0"
crop[w] | "360.99960000000004"
crop[h] | "360.99960000000004"

Response:
HTTP Status: 400 Bad request
Payload:
{
   "data":{
      "message":"Si \u00e8 verificato un errore. Contatta il tuo amministratore."
   }
}
```
</details>

**Comment by asheroto** [comment]
This is happening for me as well...

https://<server>/avatar/cropped

returns status code 400 Bad request

**Comment by arielnmz** [comment]
I had a similar issue, see my comments here: https://github.com/nextcloud/server/issues/8984

It started off as a 400 error in version 21 and I started getting a 500 after upgrading to version 22.

**Comment by asheroto** [comment]
Thanks @arielnmz... funny enough my images actually exist in that directory yet I am still facing the same issue. It's my own fault... it worked until I switched the storage to S3.

![image](https://user-images.githubusercontent.com/49938263/154822017-b32fe895-a703-407e-8ff2-8851b6d6dd88.png)

**Comment by arielnmz** [comment]
Yes, the culprit seems to be the cropper plugin, that fails to load the image and sends crop coords of 0,0,0,0

https://github.com/nextcloud/server/blob/2e1491771e47eb02d297215eba6e6ce0019fc9c1/apps/settings/js/settings/personalInfo.js#L100

![image](https://user-images.githubusercontent.com/24735547/154822176-0462ff9d-21ea-417f-ae7c-0b79f7706080.png)

**Comment by szaimen** [comment]
Hi, please update to 24.0.9 or better 25.0.3 and report back if it fixes the issue. Thank you!

My goal is to add a label like e.g. 25-feedback to this ticket of an up-to-date major Nextcloud version where the bug could be reproduced. However this is not going to work without your help. So thanks for all your effort!

If you don't manage to reproduce the issue in time and the issue gets closed but you can reproduce the issue afterwards, feel free to create a new bug report with up-to-date information by following this link: https://github.com/nextcloud/server/issues/new?assignees=&labels=bug%2C0.+Needs+triage&template=BUG_REPORT.yml&title=%5BBug%5D%3A+

**Comment by penguin86** [comment]
I confirm is fixed for me on 25.0.3. Thank you!

**Comment by szaimen** [comment]
Thanks for verifying!
